jaxrs-server/src/main/java/io/micronaut/jaxrs/runtime/core/JaxRsExceptionHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jaxrs.runtime.core;
+
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.server.exceptions.ExceptionHandler;
+import jakarta.inject.Singleton;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+/**
+ * Handles JAX-RS exceptions that occur during the execution of an HTTP request.
+ *
+ * @author guillermocalvo
+ */
+@Singleton
+@Primary
+public class JaxRsExceptionHandler
+    implements ExceptionHandler<WebApplicationException, HttpResponse<?>> {
+
+    @Override
+    public HttpResponse<?> handle(HttpRequest request, WebApplicationException exception) {
+        final Response body = exception.getResponse();
```

Comment:
Should not we be using the `ErrorResponse` API ?

See example in [Custom Exception Handler](https://docs.micronaut.io/latest/guide/#customExceptionHandler)

Reply:
`WebApplicationException` may contain a payload directly set from client code. Would you rather drop it and replace it with a generic `ErrorResponse` here?

Reply:
I have changed the code to use `errorResponse` API. Please, note that the throwable which caused the handler to be executed is passed to the ErrorResponse API with the `cause` method: 

```
 return responseProcessor.processResponse(ErrorContext.builder(request)
                 .errorMessage(exception.getMessage())
                 .cause(exception)
```

Thus, the use are in total control if they want to represent the exception differently . Using the error response api allows users to use [Micronaut Problem JSON](https://micronaut-projects.github.io/micronaut-problem-json/latest/guide/) instead of vnd.error.  

https://github.com/micronaut-projects/micronaut-jaxrs/pull/195/commits/f55d76e0aff479f9ad1a78081479a575113518bd